packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart

```diff
@@ -345,6 +346,104 @@ void main() {
       );
     });
   });
+
+  group('daemon queue', () {
+    DebounceOperationQueue<int, String> queue;
+    const Duration debounceDuration = Duration(seconds: 1);
+
+    setUp(() {
+      queue = DebounceOperationQueue<int, String>();
+    });
+
+    /// Runs a callback using FakeAsync.run while continually pumping the
+    /// microtask queue. This avoids a deadlock when tests `await` a Future
+    /// which queues a microtask that will not be processed unless the queue
+    /// is flushed.
+    Future<T> runFakeAsync<T>(Future<T> Function(FakeAsync time) f) async {
+      return FakeAsync().run((FakeAsync time) async {
+        bool pump = true;
+        final Future<T> future = f(time).whenComplete(() => pump = false);
+        while (pump) {
+          time.flushMicrotasks();
+        }
+        return future;
+      }) as Future<T>;
+    }
```

Comment:
@jonahwilliams this is a workaround for the issue described https://github.com/flutter/flutter/issues/60675. I don't know if it's the best way, or if we should pull this out somewhere and make it more reusable for the other tests that might have this issue?

Reply:
Seems like a handy feature, you could definitely put it into one of the common testing libraries.

Reply:
Moved it to `test/src/common.dart` since none of the others seemed appropriate, but lmk if somewhere else (or a new one) would be better. This might make it easy to fix https://github.com/flutter/flutter/issues/60675.